### PR TITLE
account for cases when parentNode and nextNode are undefined

### PR DESCRIPTION
**Pull request by patrickkettner**
fix for #1121

### CODE REVIEW

File: packages/optimizer/spec/lib/HtmlDomHelper.test.js [hunk 1]
```diff
@@ -28,6 +28,21 @@ test('findMetaViewport returns null if tag is not present', async () => {
   expect(result).toBeNull();
 });
 
+test('works even when template is the final node', async () => {
+  let err = null;
+
+  try {
+    await treeParser.parse(`<!doctype html>
```
Comment (sebastianbenz) [inline packages/optimizer/spec/lib/HtmlDomHelper.test.js:35]: Does this test ever fail?

**Review by sebastianbenz — COMMENTED** [review]
Can we test that non of the transformers fails when skipNodeAndChildren returns null?

File: packages/optimizer/spec/lib/HtmlDomHelper.test.js (hunk 1 above)
Comment (patrickkettner) [reply packages/optimizer/spec/lib/HtmlDomHelper.test.js:35]: it would fail if the code I added isn't there

**Comment by patrickkettner** [comment]
> Can we test that non of the transformers fails when skipNodeAndChildren returns null?

will do

**Comment by patrickkettner** [comment]
@sebastianbenz I have gotten stuck on the tests. I am trying to reuse the same looped abstraction to test all of the transformers, and just unregister, mock, and reregister. But I am getting lost in jest - I pushed my current (broken) state, would you be able to take a look?

File: packages/optimizer/spec/helpers/TransformerRunner.js [hunk 2]
```diff
@@ -45,56 +46,81 @@ if (WRITE_SNAPSHOT) {
...
         }
-        done();
+      }
+      done();
+    };
+
+    getDirectories(testConfig.testDir).forEach((testDir) => {
+      it(basename(testDir), (done) => runTests(testDir, done));
+
+      it('works when skipNodeAndChildren returns null', function (done) {
```
Comment (sebastianbenz) [inline packages/optimizer/spec/helpers/TransformerRunner.js:103]: I'd prefer not to add more tests here. Instead, we could add an end-to-end test [here](https://github.com/ampproject/amp-toolbox/tree/master/packages/optimizer/spec/end-to-end) that runs a document with a template as last element through all transformers.

File: packages/optimizer/spec/lib/HtmlDomHelper.test.js (hunk 1 above)
Comment (sebastianbenz) [reply packages/optimizer/spec/lib/HtmlDomHelper.test.js:35]: ok, interesting. skipNodeAndChildren is only called from within transformers and these shouldn't be run when calling treeParser.parse.

File: packages/optimizer/spec/helpers/TransformerRunner.js (hunk 2 above)
Comment (patrickkettner) [reply packages/optimizer/spec/helpers/TransformerRunner.js:103]: sounds good

File: packages/optimizer/package.json [hunk 3]
```diff
@@ -42,6 +42,7 @@
     "htmlparser2": "5.0.1",
     "https-proxy-agent": "5.0.0",
     "lru-cache": "6.0.0",
+    "mock-require": "^3.0.3",
```
Comment (sebastianbenz) [inline packages/optimizer/package.json:45]: Do we still need this?

**Review by sebastianbenz — APPROVED** [review]
Thanks!

**Comment by sebastianbenz** [comment]
Can you revert the package lock to main.

**Comment by sebastianbenz** [comment]
Thanks Patrick! The CI failures were unrelated to this PR.

**Pull request merged**